<?xml version="1.0"?>
<launch>

	<!--user configure parameters for ros start-->
	<arg name="lvx_file_path" default="livox_test.lvx"/>
	<arg name="bd_list" default="100000000000000"/>
	<arg name="xfer_format" default="1"/>
	<arg name="multi_topic" default="0"/>
	<arg name="data_src" default="0"/>
	<arg name="publish_freq" default="10.0"/>
	<arg name="output_type" default="0"/>
	<arg name="rviz_enable" default="false"/>
	<arg name="rosbag_enable" default="false"/>
	<arg name="cmdline_arg" default="$(arg bd_list)"/>
	<arg name="msg_frame_id" default="livox_frame"/>
	<arg name="lidar_bag" default="false"/>
	<arg name="imu_bag" default="false"/>
	<arg name="sample_at_startup" default="false"/>
	<arg name="config_file" default="$(find livox_ros_driver2)/config/MID360_config.json"/>
	<arg name="enable_filter" default="true"/>
	<arg name="box_frame" default="aristos_base_footprint"/>
	<arg name="box_min_x" default="-1.0"/>
	<arg name="box_min_y" default="-1.0"/>
	<arg name="box_min_z" default="-1.0"/>
	<arg name="box_max_x" default="1.0"/>
	<arg name="box_max_y" default="1.0"/>
	<arg name="box_max_z" default="1.0"/>
	<!--user configure parameters for ros end--> 

	<!-- Nodelet manager -->
    <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen"/>

    <!-- New nodelet -->
    <node pkg="nodelet" type="nodelet" name="livox_lidar_publisher2" args="load livox_ros_driver2/DriverNode nodelet_manager" output="screen">
        <param name="xfer_format" value="$(arg xfer_format)"/>
		<param name="multi_topic" value="$(arg multi_topic)"/>
		<param name="data_src" value="$(arg data_src)"/>
		<param name="publish_freq" type="double" value="$(arg publish_freq)"/>
		<param name="output_data_type" value="$(arg output_type)"/>
		<param name="cmdline_str" type="string" value="$(arg bd_list)"/>
		<param name="cmdline_file_path" type="string" value="$(arg lvx_file_path)"/>
		<param name="user_config_path" type="string" value="$(arg config_file)"/>
		<param name="frame_id" type="string" value="$(arg msg_frame_id)"/>
		<param name="enable_lidar_bag" type="bool" value="$(arg lidar_bag)"/>
		<param name="enable_imu_bag" type="bool" value="$(arg imu_bag)"/>
		<param name="sample_at_startup" type="bool" value="$(arg sample_at_startup)"/>
    </node>

	<node pkg="nodelet" type="nodelet" name="custom_livox_filter" args="load livox_ros_driver2/CustomFilterNodelet nodelet_manager" output="screen">
		<remap from="input_point_cloud" to="livox_lidar_publisher2/livox/lidar"/>
		<remap from="filtered_point_cloud" to="/livox/lidar"/>
		<param name="enable" type="bool" value="$(arg enable_filter)"/>
		<param name="box_frame" type="string" value="$(arg box_frame)"/>
		<param name="box_min_x" type="double" value="$(arg box_min_x)"/>
		<param name="box_min_y" type="double" value="$(arg box_min_y)"/>
		<param name="box_min_z" type="double" value="$(arg box_min_z)"/>
		<param name="box_max_x" type="double" value="$(arg box_max_x)"/>
		<param name="box_max_y" type="double" value="$(arg box_max_y)"/>
		<param name="box_max_z" type="double" value="$(arg box_max_z)"/>
	</node>

	<group if="$(arg rviz_enable)">
		<node name="livox_rviz" pkg="rviz" type="rviz" respawn="true"
				args="-d $(find livox_ros_driver2)/config/display_point_cloud_ROS1.rviz"/>
    </group>


</launch>
